liquidity/liquidity.go

```diff
@@ -841,12 +823,34 @@ func (m *Manager) SuggestSwaps(ctx context.Context, autoloop bool) (
 // suggestSwap checks whether we can currently perform a swap, and creates a
 // swap request for the rule provided.
 func (m *Manager) suggestSwap(ctx context.Context, traffic *swapTraffic,
-	balance *balances, rule *TypedRule, restrictions *Restrictions,
+	balance *balances, rule *TypedRule, outRestrictions *Restrictions,
 	autoloop bool) (swapSuggestion, error) {
 
+	var (
+		builder      swapBuilder
+		restrictions *Restrictions
+	)
+
+	switch rule.Type {
+	case swap.TypeOut:
+		builder = newLoopOutBuilder(m.cfg)
+		restrictions = outRestrictions
```

Comment:
nit: I'd keep the incoming `outRestrictions` arg with it's old name (`restrictions`) instead.